Dockerfile

```diff
@@ -1,38 +1,36 @@
-FROM ruby:2.2.3-slim
+FROM ruby:2.3.1-slim
 
-RUN apt-get update && apt-get install -y wget apt-transport-https git
-RUN wget -qO- https://deb.nodesource.com/gpgkey/nodesource.gpg.key | apt-key add -
-RUN echo 'deb https://deb.nodesource.com/node_0.12 jessie main' > /etc/apt/sources.list.d/nodesource.list
-RUN apt-get update && apt-get install -y nodejs build-essential
+RUN curl -sL https://deb.nodesource.com/setup_6.x | bash -
+RUN apt-get update && apt-get install -y nodejs build-essential libmysqlclient-dev libpq-dev libsqlite3-dev git
+RUN npm install -g npm
 
 WORKDIR /app
 
 # Mostly static
-ADD config.ru /app/
-ADD Rakefile /app/
-ADD bin /app/bin
-ADD public /app/public
-ADD db /app/db
-ADD .env.bootstrap /app/.env
-ADD .ruby-version /app/.ruby-version
+COPY config.ru /app/
```

Comment:
`ADD` can be surprising because it will decompress tarballs into the layer. [`COPY` is preferred](https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#/add-or-copy)